### PR TITLE
tutorial-screen

### PR DESCRIPTION
**Pull request by at-khoanguyen**

### CODE REVIEW

File: Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                homeButton.setTitle("Home", for: .normal)
+            } else {
+                homeButton.setTitle("Next", for: .normal)
+            }
+        }
+    }
+
+    @IBAction func nextButtonTouchUpInside(_ sender: UIButton) {
+        let currentPage = getIndex()
+//        print(currentPage)
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift]: Remove this line.

File: Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if sender.titleLabel?.text == "Home" {
+            navigationController?.pushViewController(homeViewController(), animated: true)
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift]: Define a new function to navigate a new screen.

File: Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            return
+        }
+        if currentPage < photosCount - 1 {
+            scrollView.setContentOffset(CGPoint(x: CGFloat(currentPage + 1) * scrollView.bounds.size.width, y: 0), animated: true)
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift]: Create a new variable with `CGPoint` type. It make this line code shorter and more readable

File: Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+    }
+
+    @IBAction func skipButtonTouchUpInside(_ sender: UIButton) {
+        let homeVC = homeViewController()
+        navigationController?.pushViewController(homeVC, animated: true)
+
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift]: Remove empty line

File: Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        scrollView.delegate = self
+        navigationController?.navigationBar.isTranslucent = false
+    }
+
+    override func didReceiveMemoryWarning() {
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/TutorialViewController.swift]: Remove if no using

File: Tutorial screen/tutorial screen/tutorial screen/homeViewController.swift [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+//  homeViewController.swift
+//  tutorial screen
+//
+//  Created by Khoa Nguyen D.B. on 11/29/17.
+//  Copyright © 2017 Bao Khoa. All rights reserved.
+//
+
+import UIKit
+
+class homeViewController: UIViewController {
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/homeViewController.swift]: The first character of class name is uppercase

File: Tutorial screen/tutorial screen/tutorial screen/AppDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    }()
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.backgroundColor = UIColor.white
+        window?.makeKeyAndVisible()
+        let tutorialViewController = TutorialViewController()
+
+        let navi = UINavigationController(rootViewController: tutorialViewController)
+        if (tutorialViewController.checkLog() == true) {
```
[the remark this answers is not in the document]
Comment (at-khoanguyen) [reply Tutorial screen/tutorial screen/tutorial screen/AppDelegate.swift]: Thanks a lot bro ^_^

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Tutorial/TutorialViewController.swift [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+//  TutorialViewController.swift
+//  tutorial screen
+//
+//  Created by Khoa Nguyen D.B. on 11/28/17.
+//  Copyright © 2017 Bao Khoa. All rights reserved.
+//
+
+import UIKit
+
+class TutorialViewController: UIViewController, UIScrollViewDelegate {
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Tutorial/TutorialViewController.swift]: Missing `final` keyword

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Tutorial/TutorialViewController.swift [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            } else {
+                homeButton.setTitle("Next", for: .normal)
+            }
+        }
+    }
+
+    @IBAction func nextButtonTouchUpInside(_ sender: UIButton) {
+        let currentPage = getIndex()
+        if sender.titleLabel?.text == "Home" {
+            let homeVC = HomeViewController()
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Tutorial/TutorialViewController.swift]: I think you should define a new function for navigate to Home screen.
```swift
private func gotoHomeScreen() {
   let vc = HomeViewController()
   navigationController?.pushViewController(vc, animated: true)
}
```

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Tutorial/TutorialViewController.swift [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            return
+        }
+        if currentPage < photosCount - 1 {
+            scrollView.setContentOffset(CGPoint(x: CGFloat(currentPage + 1) * scrollView.bounds.size.width, y: 0), animated: true)
+        }
+
+    }
+
+    @IBAction func skipButtonTouchUpInside(_ sender: UIButton) {
+        let homeVC = HomeViewController()
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Tutorial/TutorialViewController.swift]: Same above

File: Tutorial screen/tutorial screen/tutorial screen/AppDelegate.swift [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+        window?.backgroundColor = UIColor.white
+        window?.makeKeyAndVisible()
+
+        let vc = TutorialViewController()
+        let navi = UINavigationController(rootViewController: vc)
+
+        if Session.instance.firstTimeLaunching {
+            window?.rootViewController = navi
+            Session.instance.firstTimeLaunching = false
+
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/AppDelegate.swift]: Remove empty line

File: Tutorial screen/tutorial screen/tutorial screen/AppDelegate.swift [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+        } else {
+            window?.rootViewController = HomeViewController()
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/AppDelegate.swift]: Do you think that no need set `UINavigationController` instance for `window?.rootViewController`?

**Pull request merged**